xonsh/completer.py

```diff
@@ -201,24 +210,29 @@ def find_and_complete(self, line, idx, ctx=None):
             endidx += 1
         endidx = endidx - 1 if line[endidx] in WS else endidx
         prefix = line[begidx:endidx+1]
-        return self.complete(prefix, line, begidx, endidx, ctx=ctx), begidx, endidx
+        return (self.complete(prefix, line, begidx, endidx, ctx=ctx),
+                begidx, endidx)
 
     def _add_env(self, paths, prefix):
+        _ = self
         if prefix.startswith('$'):
             csc = builtins.__xonsh_env__.get('CASE_SENSITIVE_COMPLETIONS')
             startswither = startswithnorm if csc else startswithlow
             key = prefix[1:]
             keylow = key.lower()
-            paths.update({'$' + k for k in builtins.__xonsh_env__ if startswither(k, key, keylow)})
+            paths.update('$' + k for k in builtins.__xonsh_env__
+                         if startswither(k, key, keylow))
 
     def _add_dots(self, paths, prefix):
-        if prefix in {'', '.'}:
-            paths.update({'./', '../'})
-        if prefix == '..':
+        _ = self
```

Comment:
You're going to see this a lot in my forthcoming lint cleanup changes since there are a large number of class methods that do not depend on the state of an instance of the class.